crates/observe/src/tracing_warp.rs

```diff
@@ -0,0 +1,26 @@
+use {
+    crate::tracing::HeaderExtractor,
+    opentelemetry::global,
+    tracing_opentelemetry::OpenTelemetrySpanExt,
+    warp::http::HeaderMap,
+};
+
+pub fn warp_tracing(info: warp::trace::Info) -> tracing::Span {
+    let headers: &HeaderMap = info.request_headers();
+
+    // Extract OTEL context from headers
+    let parent_cx = global::get_text_map_propagator(|prop| prop.extract(&HeaderExtractor(headers)));
+    let request_id = headers
+        .get("X-Request-Id")
+        .and_then(|x| x.to_str().ok())
+        .unwrap_or("");
```

Comment:
Do we also propagate this request id to downstream requests? I'm not sure if the downstream tracing only works with the otel related headers or it also works with this header.